articles/contributing/style-guide.md

```diff
@@ -0,0 +1,451 @@
+Making use of documentation comments can thus help users to make sense of code by providing a useful reference for details that are not easily expressed using the other conventions in this document.
+
+<div class="nextstepaction">
+    [Documentation comment syntax reference](xref:microsoft.quantum.qsharp-ref.statements#documentation-comments)
+</div>
+
+In order to effectively use this functionality to help users, we recommend keeping a few things in mind as you write documentation comments.
+
+# [Guidance](#tab/guidance)
+
+We suggest:
+
+- Each public function, operation, and user-defined type should be immediately preceded by a documentation comment.
+- At a minimum, each documentation comment should include the following sections:
+    - Summary
+    - Input
+    - Output (if applicable)
+- Ensure that all summaries are two sentences or less. If more room is needed, provide a `# Description` section immediately following `# Summary` with complete details.
+- Where reasonable, do not include math in summaries, as not all clients support TeX notation in summaries. Note that when writing prose documents (e.g. TeX or Markdown), it may be preferable to use longer line lengths.
+- Provide all relevant mathematical expressions in the `# Description` section.
+- When describing inputs, do not repeat the types of each input as these can be inferred by the compiler and risk introducing inconsistency.
+- Provide examples as appropriate, each in their own `# Example` section.
+- Briefly describe each example before listing code.
+- Cite all relevant academic publications (e.g.: papers, proceedings, blog posts, and alternative implementations) in a `# References` section as a bulleted list of links.
+- Ensure that, where possible, all citation links are to permanent and immutable identifiers (DOIs or versioned arXiv numbers).
+- When an operation or function is related to other operations or functions by functor variants, list other variants as bullets in the `# See Also` section.
+- Leave a blank comment line between level-1 (`/// #`) sections, but do not leave a blank line between level-2 (`/// ##`) sections.
+
+# [Examples](#tab/examples)
+
+#### ☑ ####
+
+```
+/// # Summary
+/// Applies a rotation about the X-axis by a given angle.
+///
+///
+/// # Description
+/// This operation rotates a single qubit by the unitary operation
+/// \begin{align}
+///     R_x(\theta) \mathrel{:=} e^{-i \theta \sigma_x / 2}.
+/// \end{align}
+///
+/// # Input
+/// ## theta
+/// Angle about which the qubit is to be rotated.
+/// ## qubit
+/// Qubit to which the gate should be applied.
+///
+/// # Remarks
+/// Equivalent to:
+/// ```qsharp
+/// R(PauliX, theta, qubit);
+/// ```
+///
+/// # See Also
+/// - Ry
+/// - Rz
+operation Rx(theta : Double, qubit : Qubit) : Unit {
+    body (...) { R(PauliX, theta, qubit); }
+    adjoint (...) { R(PauliX, -theta, qubit); }
+    controlled auto;
+    controlled adjoint auto;
+}
+```
+
+***
+
+## Formatting Conventions ##
```

Comment:
Not sure if you want to mention the formatter here. Granted, it has some limitations related to line breaks and whitespace. And we may also want to wait until we have integrated it better with the editor. It just seems a bit weird to have a Formatting section and not mention the operation to format Q#.

Reply:
My thought was for the style guide to be somewhat independent of editor tooling, but I agree it's relevant here so I'll revise accordingly.